hooks/gen_docs.py

```diff
@@ -1,10 +1,263 @@
 """Generates the whole 'generatable' KPOps documentation"""
+import os
 import subprocess
+from pathlib import Path
+
+import yaml
 
 from hooks import PATH_ROOT
+from kpops.cli.registry import _find_classes
+from kpops.components.base_components.pipeline_component import PipelineComponent
+from kpops.utils.yaml_loading import load_yaml_file
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+PATH_DOCS_COMPONENTS_DEPENDENCIES = (
+    PATH_DOCS_COMPONENTS / "dependencies/pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS = (
+    PATH_DOCS_COMPONENTS / "dependencies/defaults_pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_KPOPS_STRUCTURE = PATH_DOCS_COMPONENTS / "dependencies/kpops_structure.yaml"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+KPOPS_COMPONENTS_INHERITANCE_REF: dict = {
```

Comment:
either add complete type annotations or remove it and let the type checker infer it (latter is preferred if possible)